gui-tk/csvImport.tcl

```diff
@@ -275,62 +278,37 @@ proc csvWebImportDialogOK {} {
 }
 
 # Allow user to input the full web address of a desired file to opened
-proc inputUrl {} {
-	set url "http://www.patreon.com/hpcoder"
-    toplevel .csvImportURL
-    pack [entry .csvImportURL.e -textvariable $url]
-    pack [button .csvImportURL.b -text "OK" -command {destroy .csvImportURL}]
-    bind .csvImportURL <Return> {.csvImportURL.b invoke}
-    focus .csvImportURL.e
+proc inputUrl {initUrl} {
+    set x [get_pointer_x .wiring.canvas]
+    set y [get_pointer_y .wiring.canvas]	
+	set url $initUrl
+	textEntryPopup .inputUrl $url {inputOK $url}
+	.inputUrl.entry configure -textvariable url -takefocus 1	
+    wm geometry .inputUrl "+[winfo pointerx .]+[winfo pointery .]"    
     return $url
```

Comment:
You probably want to return the URL typed by the user, rather than the name of the url variable, which may require double indirection: [set [set url]] (I suspect $$url or ${$url} may not work, but perhaps it does).

Note, on line 286, the url text variable is a global variable, not the local url variable accessed elsewhere. You can declare a variable global with the global statement. TCL variable scoping is a bit of a mess, actually.

Reply:
This wasn't working because I forgot to add a line "set url [$win.entry get]"